src/libraries/System.Private.CoreLib/src/System/Math.cs

```diff
@@ -151,16 +151,8 @@ internal static void ThrowNegateTwosCompOverflow()
             throw new OverflowException(SR.Overflow_NegateTwosCompNum);
         }
 
-        internal static unsafe ulong BigMul(uint a, uint b)
+        internal static ulong BigMul(uint a, uint b)
         {
-#if TARGET_32BIT
-            if (Bmi2.IsSupported)
-            {
-                uint low;
-                uint high = Bmi2.MultiplyNoFlags(a, b, &low);
-                return ((ulong)high << 32) | low;
-            }
-#endif
```

Comment:
AFAIR, the intent was for us to have an overload that returns a tuple and uses the JIT multi-reg return hookups, but we haven't exposed that yet.

We ideally aren't removing opts like this, especially when even with the memory access it is potentially faster or close enough in most cases.

Reply:
Ok I added it back, but it contributed to more than 20% slower division (more than 10ns for 1/3) so I removed a of calls and replaced it with the fallback path `(ulong)(uint)a * (uint)b` which the JIT recognize and generates optmized code for.

Since the JIT already has specialized codegen for `32 * 32 => 64bit` division would it not make sense to just emit `mulx` there (or for normal multiplication as well) if it could lead to more efficient code (maybe when writing to memory or when the register usage gets better). 

It seems clang uses `mulx` for  `32 * 32 => 64bit` in 32bit mode and  `64 * 64 => 128bit` in 64bit mode.